atom/index.test.ts

```diff
@@ -424,3 +424,32 @@ test('can use previous value in subscribers', () => {
   unbind()
   clock.runAll()
 })
+test('notifies the subscribed listener with current and old values for a store that had an initial value', () => {
+  let events: (number | undefined)[] = []
+  let $store = atom<number>(1)
+
+  $store.subscribe((value, oldValue?: number) => {
```

Comment:
You need to have here `listen`.

`listen` will be called on changes, when `subscribe` is called on changes and on initial state (where `oldValue` will be `undefined` of course).

Reply:
I don't understand. Why would it be `undefined` if the initial store value is `1`?

Reply:
Ah, I think I am starting to understand. At the time the subscription is established there is no change in value and the handler just receives the current value.

Reply:
Maybe we have issue. Let’s change test a little:

1. `events.push(oldValue, value)` → `events.push([oldValue, value])` so we will see pairs.
2. `equal(events, [1, 2])` → `equal(events, [[1, undefined], [1, 2]])`
3. The same with other tests and checks